traitlets/utils/tests/test_importstring.py

```diff
@@ -0,0 +1,35 @@
+# encoding: utf-8
+# Copyright (c) IPython Development Team.
+# Distributed under the terms of the Modified BSD License.
+#
+# Adapted from enthought.traits, Copyright (c) Enthought, Inc.,
+# also under the terms of the Modified BSD License.
+"""Tests for traitlets.utils.importstring."""
+
+import os
+from unittest import TestCase
+
+from ..importstring import import_item
+
+
+class TestImportItem(TestCase):
+
+    def test_import_unicode(self):
+        self.assertIs(os, import_item(u'os'))
+        self.assertIs(os.path, import_item(u'os.path'))
+        self.assertIs(os.path.join, import_item(u'os.path.join'))
+
+    def test_import_bytes(self):
+        self.assertIs(os, import_item(b'os'))
```

Comment:
It seems odd to me to accept `bytes` or `unicode` in PY2 but not in PY3.  Having the behavior of `import_item` be that it consistently accepts string-like types felt like the cleanest way to do this to me.